
import "stage.proto";
import "common.proto";

//请求关卡信息返回 command:901
message StageInfoResponse{
	repeated Stage stage = 1;
	required int32 elite_stage_times = 2;
	required int32 act_stage_times = 3;
	required int32 sweep_times = 4;

}

//请求章节奖励信息返回 command:902
message ChapterInfoResponse{
	repeated StageAward stage_award = 1;
}

// 进入关卡战斗返回 command:903
message StageStartResponse{
	required CommonResponse res = 1;
	optional int32 drop_num = 2;  //掉落数量
	repeated BattleUnit red = 4; //红方数据 自己
	repeated BattleUnitGrop blue = 5; //对方数据
	optional BattleUnit friend = 6; // 好友
	optional int32 hero_unpar = 7; // 无双
	optional int32 hero_unpar_level = 8; // 无双
	optional int32 monster_unpar = 9; // 怪物无双
	optional BattleUnit replace = 10; // 乱入
	optional int32 replace_no = 11;
	repeated BattleUnit awake = 12; //
	repeated int32 awake_no = 13;
	optional int32 seed1= 14;
	optional int32 seed2= 15;
}

// 关卡结算返回 command:904
message StageSettlementResponse{
    required CommonResponse res = 1;
	optional GameResourcesResponse drops = 2;
}

// 请求队形信息: 905
message FormationResponse{
	repeated LineUp lineup = 1; // 队形
}

// 请求无双信息：906
message UnparalleledResponse{
	repeated Unparalleled unpar =  1; //无双信息
}

// 关卡扫荡返回 command:907
message StageSweepResponse{
    required CommonResponse res = 1;
	optional GameResourcesResponse drops = 2;
}
